hack/generated/pkg/genruntime/conditions/conditions.go

```diff
@@ -0,0 +1,166 @@
+	// For conditions which have positive polarity (Status == True is their normal/healthy state), this will set when Status == True
+	// For conditions which have negative polarity (Status == False is their normal/healthy state), this will be set when Status == False.
+	// Conditions in Status == Unknown always have a severity of None as well.
+	// This is the default state for conditions.
+	ConditionSeverityNone ConditionSeverity = ""
+)
+
+type ConditionType string
+
+const (
+	// ConditionTypeReady is a condition indicating if the resource is ready or not.
+	// A ready resource is one that has been successfully provisioned to Azure according to the
+	// resource Spec.
+	ConditionTypeReady = "Ready"
+)
+
+var _ fmt.Stringer = Condition{}
+
+// Condition defines an extension to status (an observation) of a resource
+type Condition struct {
+	// Type of condition.
+	// +kubebuilder:validation:Required
+	Type ConditionType `json:"type"`
+
+	// Status of the condition, one of True, False, or Unknown.
+	// +kubebuilder:validation:Required
+	Status metav1.ConditionStatus `json:"status"`
+
+	// Severity with which to treat failures of this type of condition.
+	// For conditions which have positive polarity (Status == True is their normal/healthy state), this will be omitted when Status == True
+	// For conditions which have negative polarity (Status == False is their normal/healthy state), this will be omitted when Status == False.
+	// This is omitted in all cases when Status == Unknown
+	// +kubebuilder:validation:Optional
+	Severity ConditionSeverity `json:"severity,omitempty"`
+
+	// LastTransitionTime is the last time the condition transitioned from one status to another.
+	// +kubebuilder:validation:Required
+	LastTransitionTime metav1.Time `json:"lastTransitionTime"`
+
+	// Reason for the condition's last transition.
+	// Reasons are upper CamelCase (PascalCase) with no spaces. A reason is always provided, this field will not be empty.
+	// +kubebuilder:validation:Required
+	Reason string `json:"reason"`
+
+	// Message is a human readable message indicating details about the transition. This field may be empty.
+	// +kubebuilder:validation:Optional
+	Message string `json:"message,omitempty"`
+}
+
+// IsEquivalent returns true if this condition is equivalent to the passed in condition.
+// Two conditions are equivalent if all of their fields EXCEPT LastTransitionTime are the same.
+func (c Condition) IsEquivalent(other Condition) bool {
+	return c.Type == other.Type &&
+		c.Status == other.Status &&
+		c.Severity == other.Severity &&
+		c.Reason == other.Reason &&
+		c.Message == other.Message
+}
+
+// Copy returns an independent copy of the Condition
+func (c Condition) Copy() Condition {
+	return c
+}
+
+// String returns a string representation of this condition
+func (c Condition) String() string {
+	return fmt.Sprintf(
+		"Condition [%s], Status = %q, Severity = %q, Reason = %q, Message = %q, LastTransitionTime = %q",
```

Comment:
Is `%q` correct for a `metav1.Time` type?

Reply:
Yes, `metav1.Time` embeds `time.Time`, and `time.Time` has a [String()](https://pkg.go.dev/time#Time.String) function which renders text that looks like: `2021-08-02 10:18:10.230766841 -0700 PDT`. So `%q` is just wrapping that in quotes.